### ISSUE TITLE
`UP008` does not trigger on `builtins.super`

### ISSUE THREAD

**Issue by robsdedude** [astral-sh/ruff#18689]
### Summary

While looking at https://github.com/astral-sh/ruff/pull/18688, I realized that `UP008` outright ignores `builtins.super`. I'm pretty sure it shouldn't, should it?

```python
import builtins


class C:
    def f(self):
        builtins.super(C, self).f()  # doesn't get linted
```

**Comment by LaBatata101** [comment]
Weird, I thought `match_builtin_expr` handled the `builtins` module case

**Comment by LaBatata101** [comment]
I've updated the PR to fix this case.
